dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/RateLimiter.java

```diff
@@ -0,0 +1,45 @@
+package com.salesforce.dockerfileimageupdate.utils;
+
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Bucket;
+import io.github.bucket4j.Refill;
+import java.time.Duration;
+
+//based on token-bucket algorithm
+public class RateLimiter {
+
+    private long rateLimit;
```

Comment:
This should be `final`, I don't think we want to ever change the rate limit, right?